models/asset_stat/asset_stat.go

```diff
@@ -1,34 +1,33 @@
 package assetStat
 
 import (
-	"database/sql"
-	"github.com/jmoiron/sqlx"
+	"github.com/cndy-store/analytics/utils/sql"
 	"github.com/stellar/go/clients/horizon"
 	"log"
 	"time"
 )
 
 type AssetStat struct {
-	Id          *uint32    `db:"id"           json:"-"`
-	PagingToken *string    `db:"paging_token"   json:"paging_token,omitempty"`
+	PagingToken *string    `db:"paging_token" json:"paging_token,omitempty"`
 	AssetType   *string    `db:"asset_type"   json:"asset_type,omitempty"`
 	AssetCode   *string    `db:"asset_code"   json:"asset_code,omitempty"`
 	AssetIssuer *string    `db:"asset_issuer" json:"asset_issuer,omitempty"`
-	TotalAmount *string    `db:"total_amount" json:"total_amount,omitempty"`
+	TotalAmount *int64     `db:"total_amount" json:"total_amount,omitempty"`
 	NumAccounts *int32     `db:"num_accounts" json:"num_accounts,omitempty"`
 	NumEffects  *int32     `db:"num_effects"  json:"num_effects,omitempty"`
 	CreatedAt   *time.Time `db:"created_at"   json:"created_at,omitempty"`
 }
 
-func New(db *sqlx.DB, effect horizon.Effect, timestamp time.Time) (err error) {
+func New(db interface{}, effect horizon.Effect, timestamp time.Time) (err error) {
```

Comment:
Could you explain why changing this type to `interface{}` is necessary?

Reply:
This enables `New()` to be called with either `*sqlx.DB` or `*sqlx.TX` (which is the handle for a statement inside a transaction. The abstraction can also be found in `utils/sql/sql.go`.

I'm not sure whether sqlx provides an abstract interface for the `DB` and `TX` types, if so, this could also be used instead of the generic `interface{}`.

Reply:
Can't find anything in https://github.com/jmoiron/sqlx/blob/master/sqlx.go